features/cellular/framework/device/CellularContext.cpp

```diff
@@ -69,6 +69,10 @@ CellularContext::CellularContext() : _next(0), _stack(0), _pdp_type(DEFAULT_PDP_
 
 void CellularContext::cp_data_received()
 {
+    if (!_cp_netif) {
+        tr_warn("Cellular Non-IP callback missing");
```

Comment:
Maybe tr_error is better here or maybe even an assert.

Reply:
It's actually fine (but not desirable) to receive callback before `_cp_netif` is created. E.g. `QUECTEL_BG96_CellularContext::urc_nidd` is subscribed already at its constructor, but `_cp_netif` create is deferred until `CellularNonIPSocket::open` so the modem may be sending events to a previous NIDD context.